### ISSUE TITLE
False E1111 on threading.Lock.acquire

### ISSUE THREAD

**Issue by CLevasseur** [PyCQA/pylint#3212]
### Steps to reproduce
```python
from threading import Lock
_ = Lock().acquire()
```

Run pylint on this script

### Current behavior
```bash
$ pylint bug_pylint.py
************* Module bug_pylint
bug_pylint.py:3:0: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

### Expected behavior
No `E1111` should be raised since `Lock.acquire` returns a `bool`

### pylint --version output
```bash
$ pylint --version
pylint 2.4.3
astroid 2.3.2
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 05:52:31)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]
```

**Comment by svenpanne** [comment]
Just for clarification: If I see this correctly, there is still no released version of `pylint` containing that fix in `astroid`, correct? `astroid` needs a new release plus `pylint`, too, if I'm not mistaken.

**Comment by PCManticore** [comment]
Yes, this will be part of the new minor `astroid` release.
